assets/js/modules/reader-revenue-manager/components/common/PublicationSelect.js

```diff
@@ -0,0 +1,164 @@
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { MODULES_READER_REVENUE_MANAGER } from '../../datastore/constants';
+import { Option, ProgressBar, Select } from 'googlesitekit-components';
+import { isValidPublicationID } from '../../utils/validation';
+import { useSelect, useDispatch } from 'googlesitekit-data';
+
+/**
+ * Publication Select component.
+ * This component is responsible for rendering the publication select dropdown.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Object}   props                 Component props.
+ * @param {boolean}  props.isDisabled      Whether the select is disabled.
+ * @param {boolean}  props.hasModuleAccess Whether the user has access to the module.
+ * @param {string}   props.className       Additional class names to apply to the select.
+ * @param {Function} props.onChange        Callback function to run when the select value changes.
+ * @return {JSX.Element}                   Publication Select component.
+ */
+export default function PublicationSelect( props ) {
+	const {
+		isDisabled,
+		hasModuleAccess,
+		className,
+		onChange = () => {},
+	} = props;
+
+	const { setPublicationID, setPublicationOnboardingState } = useDispatch(
+		MODULES_READER_REVENUE_MANAGER
+	);
+
+	const publicationID = useSelect( ( select ) =>
+		select( MODULES_READER_REVENUE_MANAGER ).getPublicationID()
+	);
+
+	const publications = useSelect( ( select ) =>
+		hasModuleAccess !== false && ! isDisabled
+			? select( MODULES_READER_REVENUE_MANAGER ).getPublications() || []
+			: null
+	);
+
+	const publicationsLoaded = useSelect(
+		( select ) =>
+			hasModuleAccess === false ||
+			isDisabled ||
+			select( MODULES_READER_REVENUE_MANAGER ).hasFinishedResolution(
+				'getPublications'
+			)
+	);
+
+	const onPublicationChange = useCallback(
+		( index, item ) => {
+			const newPublicationID = item.dataset.value;
+
+			const publication = publications.find(
+				// eslint-disable-next-line sitekit/acronym-case
+				( { publicationId } ) => publicationId === newPublicationID
+			);
+
+			// eslint-disable-next-line sitekit/acronym-case
+			const { publicationId, onboardingState } = publication;
+
+			// eslint-disable-next-line sitekit/acronym-case
+			setPublicationID( publicationId );
+			setPublicationOnboardingState( onboardingState );
```

Comment:
This can be simplified a bit:

```suggestion
			setPublicationID( newPublicationID );
			setPublicationOnboardingState( publication.onboardingState );
```